lib/components/user/user-account-screen.js

```diff
@@ -72,21 +66,93 @@ function cloneWithHomeAndWorkAsTopLocations (loggedInUser) {
  * This screen handles creating/updating OTP user account settings.
  */
 class UserAccountScreen extends Component {
-  _updateUserPrefs = async userData => {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      // Capture whether user is a new user at this stage, and retain that value as long as this screen is active.
+      // Reminder: When a new user progresses through the account steps,
+      // isNewUser(loggedInUser) will change to false as the database gets updated.
+      isNewUser: isNewUser(props.loggedInUser),
+
+      // Last number and last time we requested a code for (to avoid repeat SMS and not waste SMS quota).
+      lastPhoneNumberRequested: null,
+      lastPhoneRequestTime: null
+    }
+  }
+
+  _updateUserPrefs = async (userData, silentOnSucceed = false) => {
     // TODO: Change state of Save button while the update action takes place.
 
     // In userData.savedLocations, filter out entries with blank addresses.
     const newUserData = clone(userData)
     newUserData.savedLocations = newUserData.savedLocations.filter(({ address }) => address && address.length)
-    await this.props.createOrUpdateUser(newUserData)
+    await this.props.createOrUpdateUser(newUserData, silentOnSucceed)
 
     // TODO: Handle UI feedback (currently an alert() dialog inside createOrUpdateUser).
   }
 
+  /**
+   * Silently persists the user data upon accepting terms.
+   * Creating the user record before the user finishes the account creation steps
+   * is required by the middleware in order to perform phone verification.
+   *
+   * @param {*} userData The user data state to persist.
+   * @returns The new user id the the caller can use.
+   */
+  _handleCreateNewUser = async userData => {
+    await this._updateUserPrefs(userData, true)
+
+    return this.props.loggedInUser.id
```

Comment:
Another pattern of awaiting a change in application state. Move this to an action.

Reply:
Removed in https://github.com/opentripplanner/otp-react-redux/pull/224/commits/3c4802751a090c2ad4e95a887b7497476a12ba22.